packages/mobile/src/transactions/saga.ts

```diff
@@ -148,21 +152,52 @@ function* refreshRecentTxRecipients() {
     }
 
     const e164PhoneNumber = addressToE164Number[address]
-    const cachedRecipient = recipientCache[e164PhoneNumber]
-    // Skip if there is no recipient to cache or we've already cached them
-    if (!cachedRecipient || recentTxRecipientsCache[e164PhoneNumber]) {
-      continue
+    if (e164PhoneNumber) {
+      const cachedRecipient = recipientCache[e164PhoneNumber]
+      // Skip if there is no recipient to cache or we've already cached them
+      if (!cachedRecipient || recentTxRecipientsCache[e164PhoneNumber]) {
+        continue
+      }
+
+      recentTxRecipientsCache[e164PhoneNumber] = cachedRecipient
+      remainingCacheStorage -= 1
     }
-
-    recentTxRecipientsCache[e164PhoneNumber] = cachedRecipient
-    remainingCacheStorage -= 1
   }
 
   yield put(updateRecentTxRecipientsCache(recentTxRecipientsCache))
 }
 
+function* addProfile(transaction: TransferItemFragment) {
+  const address = transaction.account
+  const newProfile: AddressToRecipient = {}
+  if (transaction.type === TokenTransactionType.Received) {
```

Comment:
If we have many incoming transactions from the same address we will fetch their profile info many times. Something we could do is to add a timestamp to the recipient with the date that the data was last refreshed and only read again of more than X time has passed (an hour? a day?)

What do you think? Also, do you think it makes sense to do it in this PR or create a new issue for this?

Reply:
filed an issue! :) https://github.com/celo-org/wallet/issues/300